### PR TITLE
Granular template checks for marketplace theme validation

### PR DESCRIPTION
**Pull request by brandenrodgers**
## Description and Context

Using this PR as the base just to split up the work into separate PR's to make it more reviewable.

This adds more accurate template checking to the theme validation command. Now we're correctly checking for label and screenshot annotations depending on the template type. 

## Screenshots

image

## TODO

- [x] Add unit tests for the new template checks
- [ ] This still needs more testing. I ran it against a marketplace theme and it seems to work, but there are a lot of edge cases that I could be missing.
## Who to Notify

cc/ @KatzMitch

### CODE REVIEW

File: packages/cli/lib/validators/marketplaceValidators/theme/TemplateValidator.js [hunk 1]
```diff
@@ -4,11 +4,53 @@ const {
...
   getFileAnnotations,
-  isTemplate,
+  isCodedFile,
 } = require('@hubspot/cli-lib/templates');
 const BaseValidator = require('../BaseValidator');
+const { logger } = require('@hubspot/cli-lib/logger');
 
 const TEMPLATE_LIMIT = 50;
+const TEMPLATE_COUNT_IGNORE_LIST = ['global_partial', 'none'];
+const VALIDATIONS_BY_TYPE = {
```
Comment (drewjenkins) [inline packages/cli/lib/validators/marketplaceValidators/theme/TemplateValidator.js:14]: Very cool

File: packages/cli/lib/validators/marketplaceValidators/theme/TemplateValidator.js [hunk 2]
```diff
@@ -35,45 +82,85 @@ class TemplateValidator extends BaseValidator {
...
+              )}`
+            );
+          }
+        } else {
+          logger.debug(
+            `No template type found for ${path.relative(
+              absoluteThemePath,
+              filePath
+            )}`
+          );
```
Comment (anthmatic) [inline packages/cli/lib/validators/marketplaceValidators/theme/TemplateValidator.js:150]: Should these also be validation errors?

Comment (brandenrodgers) [reply packages/cli/lib/validators/marketplaceValidators/theme/TemplateValidator.js:150]: I'm not sure, but maybe @KatzMitch will know. In this scenario the html file will have to have a `.html` extension, `isAvailableforNewContent !== false`, and no templateType annotation. 

somewhat related, I'm currently doing nothing for html files that have no annotations at all and I'm not sure if I should be showing some sort of validation message for those.

Comment (KatzMitch) [reply packages/cli/lib/validators/marketplaceValidators/theme/TemplateValidator.js:150]: Hmm since it's just specific to the marketplace, I think we should require that all HTML files have annotations and throw an error if they don't (cc: @ckconant )

Comment (ckconant) [reply packages/cli/lib/validators/marketplaceValidators/theme/TemplateValidator.js:150]: Yeah, I agree with @KatzMitch that all html files should have annotations. The only file that has limited annotations is the `base.html` layout in [boilerplate](https://github.com/HubSpot/cms-theme-boilerplate/blob/main/src/templates/layouts/base.html#L2), which only has `templateType: none` annotated.

Comment (brandenrodgers) [reply packages/cli/lib/validators/marketplaceValidators/theme/TemplateValidator.js:150]: OK, I'll add an additional error type for this 👍

Comment (brandenrodgers) [reply packages/cli/lib/validators/marketplaceValidators/theme/TemplateValidator.js]: added in a8db57d
